src/routes/solid-start/guides/data-fetching.mdx

```diff
@@ -0,0 +1,227 @@
+---
+title: "Data Fetching"
+---
+
+This guide will walk you through the basics of data fetching in SolidStart, providing practical examples and best practices.
+
+Here's a minimal example of data fetching in SolidStart:
+
+```tsx title="src/routes/index.tsx"
+import { For } from "solid-js";
+import { query, createAsync } from "@solidjs/router";
+
+const getPosts = query(async () => {
+	const posts = await fetch("https://my-api.com/blog");
+	return await posts.json();
+}, "posts");
+
+export default function Page() {
+	const posts = createAsync(() => getPosts());
+	return (
+		<ul>
+			<For each={posts()}>{(post) => <li>{post.title}</li>}</For>
+		</ul>
+	);
+}
+```
+
+In this example, we created a [`query`](https://docs.solidjs.com/solid-router/reference/data-apis/query) and used the [`createAsync`](https://docs.solidjs.com/solid-router/reference/data-apis/create-async) primitive to access the query data inside our component.
+
+## Reference
+
+- Solid Router [`query`](https://docs.solidjs.com/solid-router/reference/data-apis/query)
+- Solid Router [`createAsync`](https://docs.solidjs.com/solid-router/reference/data-apis/create-async)
+- Solid Router [`preload`](https://docs.solidjs.com/solid-router/reference/preload-functions/preload)
+- Solid [`createResource`](https://docs.solidjs.com/reference/basic-reactivity/create-resource)
+- Solid [`<Suspense>`](https://docs.solidjs.com/reference/components/suspense)
+- Solid [`<ErrorBoundary>`](https://docs.solidjs.com/reference/components/error-boundary)
```

Comment:
I'm not entirely sure this section is necessary since each part is linked out when used

Reply:
Since not every mention of the primitives and components is linked, I believe it would be helpful to have a central location for all the links to related documentation. This way, users will always know where to find all the important links on the page, similar to what is done in [Next.js](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching#reference).

However, I understand if you feel this is unnecessary. I remove it for now.